alica_engine/src/engine/TeamObserver.cpp

```diff
@@ -73,8 +73,11 @@ bool TeamObserver::updateTeamPlanTrees()
 
     std::vector<AgentId> deactivatedAgentIds;
     bool changedSomeAgent = _tm.updateAgents(deactivatedAgentIds);
-    for (auto agent : deactivatedAgentIds) {
-        _simplePlanTrees.erase(agent);
+
+    for (auto it = _simplePlanTrees.begin(); it != _simplePlanTrees.end(); /*No increment*/) {
```

Comment:
so any time a msg from an inactive agent is received this will cause a crash right? The system load is only to trigger the timeout in the first place?

Reply:
Resolved in discussion:
Receiving a msg from an inactive agent does lead to an assertion error, when the time between receiving the msg and processing the msg is greater than the timeout